### ISSUE TITLE
incorrect result of sigmoid_cross_entropy_with_logits when input is np.inf

### ISSUE THREAD

**Issue by maybeLee** [tensorflow/tensorflow#75256]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.17.0

### Custom code

Yes

### Current behavior?

Based on the documentation: https://www.tensorflow.org/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits, this API's output should be consistent with this equation: `z * -log(sigmoid(x)) + (1 - z) * -log(1 - sigmoid(x))`. However, when running the following code:

```
import tensorflow as tf
import numpy as np

x = tf.constant(np.inf)
z = tf.constant(0.9)
res1 = tf.nn.sigmoid_cross_entropy_with_logits(labels=z, logits=x)
print(f"TF's result: {res1}")  # nan
log = tf.math.log
sigmoid = tf.math.sigmoid
res2 = z * -log(sigmoid(x)) + (1 - z) * -log(1 - sigmoid(x))
print(f"Expected result: {res2}")  # inf
```

The API's output is inconsistent with the equation's output as follows:

```
TF's result: nan
Expected result: inf
```

The possible reason is that the current implementation contains `inf - xx*inf` when `x=inf`, thus lead to `NaN`.

https://github.com/tensorflow/tensorflow/blob/e4e8ba1af3511dc749a9d9d8dbac0c4dc3c7eab5/tensorflow/python/ops/nn_impl.py#L143

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np

x = tf.constant(np.inf)
z = tf.constant(0.9)
res1 = tf.nn.sigmoid_cross_entropy_with_logits(labels=z, logits=x)
print(f"TF's result: {res1}")  # nan
log = tf.math.log
sigmoid = tf.math.sigmoid
res2 = z * -log(sigmoid(x)) + (1 - z) * -log(1 - sigmoid(x))
print(f"Expected result: {res2}")  # inf
```

**Comment by sanskarmodi8** [comment]
Hi @maybeLee,

The behavior you're observing stems from TensorFlow's internal handling of extreme values like `np.inf`. While the manual computation results in `inf`, TensorFlow applies numerically stable operations that avoid overflow/underflow issues, which can lead to NaN in such cases.

The underlying formula in TensorFlow ensures stability for large values of x, which is likely why the API returns NaN instead of inf. Though this seems inconsistent with the documented formula, TensorFlow prioritizes stability over direct computation in edge cases.

One potential workaround is to manually handle extreme values before passing them to the API to avoid getting NaN in such cases.
